compose/service.py

```diff
@@ -770,6 +765,47 @@ def parse_volume_spec(volume_config):
     return VolumeSpec(external, internal, mode)
 
 
+def build_volume_from(volume_from_spec):
+    volumes_from = []
+    if isinstance(volume_from_spec.source, Service):
+        containers = volume_from_spec.source.containers(stopped=True)
+        if not containers:
+            volumes_from = ["{}:{}".format(volume_from_spec.source.create_container().id, volume_from_spec.mode)]
+        else:
+            volumes_from = ["{}:{}".format(container.id, volume_from_spec.mode) for container in containers]
+    elif isinstance(volume_from_spec.source, Container):
+        volumes_from = ["{}:{}".format(volume_from_spec.source.id, volume_from_spec.mode)]
+    return volumes_from
+
+
+def parse_volume_from_spec(volume_from_config, get_service, client):
```

Comment:
Don't be sorry :wink: ; that's what I understood, but… I though I could do both in there, to avoid updating VolumeFromSpec afterwards. But now that I'm looking at it, it feels really bad :sweat_smile:, I'll fix that :wink:.